agent/agent.go

```diff
@@ -425,6 +427,10 @@ func (a *Agent) wait(ctx, taskCtx context.Context, tc *taskContext, heartbeat ch
 		grip.Infof("received signal from heartbeat channel: %s", tc.task.ID)
 	}
 
+	if err := tc.runOomTrackerCheck(ctx); err != nil {
```

Comment:
Some questions about this:
1. How does this behave on systems that don't have an OOM killer or that don't support dmesg?
2. Should this only run for failed tasks? From Splunk we ran about 250k tasks in 2.5 million minutes in the last 24 hours, so if we ran this on every task (and if every system supported it) and it takes a minute, we would increase costs by 10%.
3. How long do we expect this to take? I think we should create a context with timeout that we pass to it to make sure it stays within some reasonable duration, one which we can probably determine based on how long it usually takes to run.

Reply:
1. They [noop](https://github.com/mongodb/jasper/blob/6554d44dd98099f903fe1e78b8fcb088da2bd7b1/subtree_oom_default.go#L11).
2. That makes sense. WRT cost, we already run it (in pre/post commands) for every server task (I think?), so it will be a net neutral change for those tasks once the command is removed from the server's evergreen.yml. Maybe we should make it an option on the task and tell users to add it where they want it? This way we won't run it when it's not necessary (it's not something we worry about on Evergreen, for instance).
3. I spot checked some server tasks and it looks like it usually takes about 2s to do the check. Does 10s sound good?

Reply:
re2: I think it makes sense to put the opt-in logic at the project level.

Reply:
Yes, I think opt-in logic on the project level makes sense. Once this ticket is closed, could you open a server ticket for them to remove the current code and opt in?

10 seconds sounds good.

Since we'd like to know how long it takes, and since it would be nice to validate that this is running, I think it'd make sense to add an additional log line, something like "found <an|no> OOM kill in <num> seconds".

Reply:
I added a project level configuration and a 10s timeout

Reply:
> found <an|no> OOM kill in seconds

done